### PR TITLE
Add SQLite unittests for the AdminServiceStore

### PR DESCRIPTION
**Pull request by agunde406**
Also includes several fixes to the diesel operations.

New migrations for postgres were tested against a running postgres database.

### CODE REVIEW

**Review by shannynalayna — CHANGES_REQUESTED** [review]
I'm getting this warning when running `cargo test --features experimental`: 

```
warning: unused imports: `ProposedCircuit`, `ProposedNode`, `ProposedService`
   --> libsplinter/src/admin/store/diesel/mod.rs:263:9
    |
263 |         ProposedCircuit, ProposedCircuitBuilder, ProposedNode, ProposedNodeBuilder,
    |         ^^^^^^^^^^^^^^^                          ^^^^^^^^^^^^
264 |         ProposedService, ProposedServiceBuilder, ServiceBuilder, Vote, VoteRecordBuilder,
    |         ^^^^^^^^^^^^^^^
    |
    = note: `#[warn(unused_imports)]` on by default

warning: 1 warning emitted
```

**Comment by agunde406** [comment]
> I'm getting this warning when running `cargo test --features experimental`:
> 
> ```
> warning: unused imports: `ProposedCircuit`, `ProposedNode`, `ProposedService`
>    --> libsplinter/src/admin/store/diesel/mod.rs:263:9
>     |
> 263 |         ProposedCircuit, ProposedCircuitBuilder, ProposedNode, ProposedNodeBuilder,
>     |         ^^^^^^^^^^^^^^^                          ^^^^^^^^^^^^
> 264 |         ProposedService, ProposedServiceBuilder, ServiceBuilder, Vote, VoteRecordBuilder,
>     |         ^^^^^^^^^^^^^^^
>     |
>     = note: `#[warn(unused_imports)]` on by default
> 
> warning: 1 warning emitted
> ```

Whoops forgot to fix that, updated

File: libsplinter/src/admin/store/diesel/operations/list_circuits.rs [hunk 1]
```diff
@@ -231,6 +196,12 @@ where
                     })?);
                 }
 
+                ret_circuits.retain(|circuit| {
```
Comment (peterschwarz) [inline libsplinter/src/admin/store/diesel/operations/list_circuits.rs:200]: This should still be done at the database level.  The correct solution should be to omit the filters from the query if it they are empty.

**Review by peterschwarz — CHANGES_REQUESTED** [review]

File: libsplinter/src/admin/store/diesel/operations/list_circuits.rs (hunk 1 above)
Comment (agunde406) [reply libsplinter/src/admin/store/diesel/operations/list_circuits.rs]: Updated

**Review by peterschwarz — APPROVED** [review]

**Review by shannynalayna — APPROVED** [review]

**Pull request merged**